doc/memory.md

```diff
@@ -0,0 +1,17 @@
+# Memory
+
+ROHD HCL provides a generic `abstract` [`Memory`](https://intel.github.io/rohd-hcl/rohd_hcl/Memory-class.html) class which accepts a dynamic number of `writePorts` and `readPorts`, where each port is of type [`DataPortInterface`](https://intel.github.io/rohd-hcl/rohd_hcl/DataPortInterface-class.html).  A `DataPortInterface` is a simple interface with `en` and `addr` as `control` signals and `data` signal(s).  In a write interface, all signals are in the same direction.  In a read interface, the `control` signals are in the opposite direction of the `data` signal(s).
+
+## Strobes
+
+A sub-class of `DataPortInterface` is the[`StrobeDataPortInterface`](https://intel.github.io/rohd-hcl/rohd_hcl/StrobeDataPortInterface-class.html), which adds `strobe` to the `data` group of signals.  The `strobe` signal is a byte-enable signal, where each bit of `strobe` controls one byte of `data`.
+
+## Register Files
+
+A sub-class of `Memory` is the [`RegisterFile`](https://intel.github.io/rohd-hcl/rohd_hcl/RegisterFile-class.html), which inherits the same flexible interface from `Memory`.  It has a configurable number of entries via `numEntries`.
+
+The `RegisterFile` accepts strobes on writes, but not on reads.
```

Comment:
I guess it could save some depending on how it's used.  If synthesized as-is (with flops), we'd need to modify the implementation to store bytes independently so that they could be independently accessed.  I'm not that familiar if macro RFs (if mapped in the backend) often have support for a strobe for read power savings?  Do you know?

Reply:
Yes, in RF implementation they do read enable at various granularities (byte, word, cache line).

Reply:
I filed two issues tracking this:
- Read strobing: #16 
- Variable width strobes: #15 

Since these are significant additional features, I'm resolving this comment and we can add the additional features separately.